### ISSUE TITLE
invalid-overload on @overload sets over **kwargs: Unpack[<open TypedDict>]

### ISSUE THREAD

**Issue by vdusek** [astral-sh/ty#4474]
### Summary

Since 0.0.74, ty reports `invalid-overload` for every `@overload` in a set built on `**kwargs: Unpack[TD]` with an open `TD`, whenever the overloads differ in which keywords they declare. 0.0.73 accepted this, and pyright and mypy accept it today.

Minimal reproducer:

```python
from typing import Protocol, overload

from typing_extensions import NotRequired, TypedDict, Unpack


class Options(TypedDict):
    limit: NotRequired[int]


class EnqueueLinks(Protocol):
    """Takes either `selector` or `requests`, never both, plus the shared `Options`."""

    @overload
    def __call__(self, *, selector: str | None = None, **kwargs: Unpack[Options]) -> None: ...

    @overload
    def __call__(self, *, requests: list[str] | None = None, **kwargs: Unpack[Options]) -> None: ...

    def __call__(
        self,
        *,
        selector: str | None = None,
        requests: list[str] | None = None,
        **kwargs: Unpack[Options],
    ) -> None: ...
```

Invoked as `ty check` with no configuration at all:

```
error[invalid-overload]: Implementation does not accept all arguments of this overload
  --> repro.py:14:9
   |
14 |     def __call__(self, *, selector: str | None = None, **kwargs: Unpack[Options]) -> None: ...
   |         ^^^^^^^^
...
19 |     def __call__(
   |         -------- Implementation defined here
info: Implementation signature `(self, /, *, selector: str | None = None, requests: list[str] | None = None,
      limit: int = ..., **kwargs: object) -> None` is not assignable to overload signature
      `(self, /, *, selector: str | None = None, limit: int = ..., **kwargs: object) -> None`
info: parameter `requests` has an incompatible type: `object` is not assignable to `list[str] | None`

error[invalid-overload]: Implementation does not accept all arguments of this overload
  --> repro.py:17:9
info: parameter `selector` has an incompatible type: `object` is not assignable to `str | None`

Found 2 diagnostics
```

The overloads here express "either `selector` or `requests`, never both, plus a set of shared options": the overloads constrain the combination, and the TypedDict carries the shared options once instead of repeating them in every overload. With an open TypedDict I cannot write that at all now, because ty reports every overload that omits a keyword another overload declares.

### Version

0.0.74 - 0.0.78

**Comment by sharkdp** [comment]
I think the problem here is that those overloads **do not** express "either selector or requests, never both". The `Options` TypedDict is open, which means that valid instances of `Options` could have arbitrary other fields. ty therefore treats `**kwargs: Unpack[Options]` as if you had written `limit: int = ..., **kwargs: object`, where the `**kwargs: object` parameter accounts for the openness of the TypedDict.

See also: https://github.com/astral-sh/ty/issues/4212#issuecomment-5220576204

Using `closed=True` would solve your problem. Is that an option for you?

**Comment by vdusek** [comment]
@sharkdp It resolved our problem. Thank you.
